src/main/scala/encry/api/http/routes/WalletInfoApiRoute.scala

```diff
@@ -1,53 +1,278 @@
+                              settingsApp: EncryAppSettings)(implicit val context: ActorRefFactory)
   extends EncryBaseApiRoute with FailFastCirceSupport with StrictLogging {
 
-  override val route: Route = pathPrefix("wallet") { infoR ~ getUtxosR }
+  override val route: Route = pathPrefix("wallet") {
+     infoR ~ getUtxosR ~ printAddressR ~ createKeyR ~ printPubKeysR ~ getBalanceR ~ transferR ~ transferContractR ~ createTokenR ~ dataTransactionR
+  }
 
   override val settings: RESTApiSettings = restApiSettings
 
-  private def getWallet: Future[EncryWallet] = (nodeViewHolder ?
-    GetDataFromCurrentView[History, UtxoState, EncryWallet, EncryWallet](_.vault))
-    .mapTo[EncryWallet]
+  private def getWallet: Future[EncryWallet] =
+    (dataHolder ?
+      GetDataFromCurrentView[History, UtxoState, EncryWallet, EncryWallet](_.vault))
+      .mapTo[EncryWallet]
+
+  private def getAddresses: Future[String] = (dataHolder ? GetViewPrintAddress)
+    .mapTo[String]
+
+  private def createKey: Future[PrivateKey25519] = (dataHolder ? GetViewCreateKey)
+    .mapTo[PrivateKey25519]
+
+  private def pubKeys: Future[List[String]] = (dataHolder ? GetViewPrintPubKeys)
+    .mapTo[List[String]]
+
+  private def getBalance: Future[String] = (dataHolder ? GetViewGetBalance)
+    .mapTo[String]
 
   def infoR: Route = (path("info") & get) {
-    getWallet
-      .map { w =>
-        Map(
-          "balances" -> w.getBalances.map{ i =>
-            if (i._1._2 != intrinsicTokenId)
-              s"TokenID(${i._1._2}) for contractHash ${i._1._1} : ${i._2}"
-            else
-              s"TokenID(${i._1._2}) for contractHash ${i._1._1} : ${BigDecimal(i._2) / 100000000}"
-          }.asJson,
-          "utxosQty" -> Random.shuffle(w.walletStorage.getAllBoxes(1000)).length.asJson
-        ).asJson
+    getWallet.map { w =>
+      Map(
+        "balances" -> w.getBalances.map{ i =>
+          if (i._1._2 != intrinsicTokenId)
+            s"TokenID(${i._1._2}) for contractHash ${i._1._1} : ${i._2}"
+          else
+            s"TokenID(${i._1._2}) for contractHash ${i._1._1} : ${BigDecimal(i._2) / 100000000}"
+        }.asJson,
+        "utxosQty" -> Random.shuffle(w.walletStorage.getAllBoxes(1000)).length.asJson
+      ).asJson
+    }.okJson()
+  }
+
+  def printAddressR: Route = (path("addr") & get) {
+    getAddresses.map(_.asJson).okJson()
+  }
+
+  def errorR: Route = path("restricted") {
+    reject(ValidationRejection("Restricted!"))
+  }
+
+  case class LoginRequest(a:String, b:String)
+
+  def createTokenR: Route = (path("createToken") & get) {
+    parameters('fee.as[Int], 'amount.as[Long]) { (fee, amount) =>
+      (dataHolder ?
+        GetDataFromPresentView[History, UtxoState, EncryWallet, Option[Transaction]] { wallet =>
+          Try {
+            val secret: PrivateKey25519 = wallet.vault.accountManagers.head.mandatoryAccount
+            val boxes: AssetBox         = wallet.vault.walletStorage
```

Comment:
do we really need to take from db all boxes and next take .head. It'll be better to take from db boxes till first correct